packages/faceted-search/src/components/Badges/BadgeCheckboxes/BadgeCheckboxesForm.component.js

```diff
@@ -86,24 +89,49 @@ const BadgeCheckboxesForm = ({
 			checkboxes.filter(c => c.checked),
 		);
 	};
+	const onToggleAll = event => {
+		const checked = event.target.checked;
+		checkboxes.forEach(entity => (entity.checked = checked));
```

Comment:
Is it safe to update `checkboxes` ? This doesn't trigger any new render but this const is used for rendering component.
How new render is triggered ? Through "onChange" call ? If yes, it is more reliable to give some new array with updated properties

Reply:
I understand your point 
In this case it will both update checkboxes internally , and will send the updated list with a "onChange". 
Most of the time "onChange" will update the checkbox on parent component and then the parent should send props back to the children, but it's not mandatory so it could cause a desync of the rendering. So i've used the same approach as "onChangeCheckBoxes" , i agree with you it's bugging me a bit too